cmd/cspc-operator/app/handler.go

```diff
@@ -122,12 +122,14 @@ func (c *Controller) syncCSPC(cspcGot *apis.CStorPoolCluster) error {
 		return nil
 	}
 
-	cspcGot, err := c.populateVersion(cspcGot)
+	cspcObj := cspcGot
+	cspcObj, err := c.populateVersion(cspcObj)
```

Comment:
change looks good.. but, better to make populateVersion to return same object in the case of error..
that avoids lot of complex logic and probable issues